src/basic/stat-util.c

```diff
@@ -163,6 +163,24 @@ int null_or_empty_fd(int fd) {
         return null_or_empty(&st);
 }
 
+int fd_is_read_only_fs(int fd) {
+        struct statvfs st;
+
+        assert(fd >= 0);
+
+        if (fstatvfs(fd, &st) < 0)
+                return -errno;
+
+        if (st.f_flag & ST_RDONLY)
+                return true;
+
+        /* See comments in path_is_read_only_fs(). */
+        if (faccessat(fd, ".", W_OK, 0) < 0 && errno == EROFS)
```

Comment:
hmm, this check will only work if `fd` refers to a directory. I think we should make it work in any case, i.e. also when pointint to a directory.

Hence maybe use the access_fd() helper instead?